conformance/tests/httproute-cors.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: cors
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: same-namespace
+  rules:
+  - filters:
+    - type: CORS
+      cors:
+        allowOrigins:
+        - "https://www.foo.com"
+        - "https://*.bar.com"
+        allowMethods:
+        - GET
+        - POST
+        - PUT
+        - PATCH
+        - OPTIONS
+        allowHeaders:
+        - "x-header-1"
+        - "x-header-2"
+        exposeHeaders:
+        - "x-header-3"
+        - "x-header-4"
+        allowCredentials: true
+        maxAge: 3600
+    backendRefs:
```

Comment:
IMO I would add an additional route with `allowCredentials: false` and e.g., `allowMethods: ["*"]`